department-of-veterans-affairs/vets-saml-proxy
# Api 3592 static token

Enables the use of static tokens to be retrieved during the refresh flow. 
To test the new functionality
-  pre-populate with 
`aws dynamodb put-item \
--table-name StaticTokens  \
--endpoint-url http://localhost:8000 \
--item \
    "{\"static_access_token\": {\"S\": \"$ACCESS_TOKEN\"}, \"static_refresh_token\": {\"S\": \"$REFRESH_TOKEN\"}, \"static_scopes\": {\"S\": \"$SCOPES\"}, \"static_expires_in\": {\"N\": \"3600\"}, \"static_icn\": {\"S\": \"$ICN\"}}"`
- exercise the refresh flow using the refresh token from above.

Note that in order for this functionality to be available an MR that creates a new dynamo table  has to be serviced. If the table is not available, the code will retain the old behavior.
